Week 2/Homework2.html

```diff
@@ -0,0 +1,128 @@
+<html>
+	<head>
+		<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
+	</head>
+	<body>
+		<br>
+		Book: <input type="text" id="book"/> <br>
+		Author: <input type="text" id="bookauthor"/> <br>
+		Price: <input type="text" id="bookprice"/> <br>
+		<button onclick="submitBook();">Submit Book</button> <br>
+		<br>
+		Library <input type="text" id="library"/> <br>
+		Place <input type="text" id="libraryplace"/> <br>
+		Books <input type="text" id="librarybooks"/> <br>
+		<button onclick="submitLibrary();">Submit Library</button>
+		<br>
+		<button onclick="output();">UpdateList</button>
+		<br>
+		<span id="libraryoutput">LIBRARY <br></span>
+		<br>
+		<span id="bookoutput">BOOK<br><br></span>
+		
+	</body>
+	
+	<script src="https://cloudboost.io/js-sdk/cloudboost.js"></script>
+	<script>
+	CB.CloudApp.init('njoumzczehoq', 'b7cc60b7-a1ad-4a0e-8d1c-aceab6dcb4f4');
+	let thelibrary = new CB.CloudObject("Library");
+	let thebooks = new CB.CloudObject("Book");
+		
+	class Library {
+		constructor(name, place) {
+			this._name = name;
+			this._place = place;
+			this._books = [];
+			thelibrary.set("library", this._name);
+			thelibrary.set("libraryplace", this._place);
+			thelibrary.save({
+				success : (res) => {
+					console.log(res);
+				}	
+			});
+	
+		}
+	}
+	class Book {
+		constructor(name, author, price, bool) {
+			let thelibrary = thebooks;
+			this._name = name;
+			this._author = author;
+			this._price = price;
+			console.log(`${name} ${author} ${price}`);
+			if(bool){
+				thelibrary.set("book", this.name);
+				thelibrary.set("bookauthor", this.author);
+				thelibrary.set("bookprice", this.price);
+				thelibrary.save({
+					success: (res) => {
+						console.log(res);
+					}
+				});
+			}else {
+				console.log("Did nothing");
+			}
+		}
+		get name (){
```

Comment:
If you aren't really doing anything special with the setter / getters, you can just use the defaults.

```js
class Book {
  constructor(name) {
    this.name = name;
  }
}
```

Behind the scenes that actually creates this...

```js
class Book {
  constructor(name) {
    this.name = name; // this calls `set name`
  }

  // when you create your own set / get, you are actually doing method overriding.
  set name(name) {
    this._name = name;
  }

  get name() {
    return this.name;
  }
}
```

Does that make sense? The idea is to use the custom setters / getters with the cloudBoost object. You can set the cloudBoost object and get the cloudBoost object data using it.